### PR TITLE
Add page about how to use Soroban contracts in Jupyter Notebooks

### LINKED ISSUE #807: Add a How-to Soroban contracts in Jupyter Notebooks page
Soroban contracts as of SDK v21.3.0 work inside Jupyter Notebooks with the `excvr` Rust Jupyter kernel.

I'm personally not a big user of Jupyter, but I know some folks are, and use them for creating interactive tutorials, demos, doing a mix of graphing and code.

I think it'd be worth adding a how-to to set it up to the docs. It doesn't need to be in a prime location in the docs, but just somewhere discoverable via search so if someone was a Jupyter fan and was searching for how to hack on contracts in Jupyter Notebooks they'd find it.

Related conversations:
- https://discord.com/channels/897514728459468821/1263811925813366794/1263811925813366794
- https://stellarfoundation.slack.com/archives/C06KTGUULUF/p1721387447128529

Example:
notebook

cc @ElliotFriend

### PR DESCRIPTION
**Pull request by leighmcculloch**
### What
Add page about how to use Soroban contracts in Jupyter Notebooks.

### Why
Jupyter Notebooks support Rust, albeit experimental, and can run Soroban contracts. For some people Jupyter Notebooks is a useful tool for experimenting, teaching, documenting, and learning.

Close #807

### CODE REVIEW

File: docs/tools/developer-tools/jupyter-notebooks.mdx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+client.add(&1, &2)
+```
+
+11. Click the play button to run the code.
+
+Congratulations you have a Jupyter Notebook with contract code that should look something like the screenshot below, ready for hacking and experimenting.
+
+## Screenshot
+
+![](jupyter-notebooks.png)
```
Comment (fnando) [inline docs/tools/developer-tools/jupyter-notebooks.mdx:66]:
```suggestion
![A running Jupyter notebook](jupyter-notebooks.png)
```

File: docs/tools/developer-tools/jupyter-notebooks.mdx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+sidebar_position: 71
+---
+
+# Jupyter Notebooks
+
+Jupyter Notebooks are a document format that can include code, text, and other rich output. And with the appropriate setup they support Soroban Rust contracts.
+
+To use Soroban contracts in a Jupyter Notebook, the following setup is required. The following setup uses Visual Studio Code, but any Jupyter client and server can be used.
+
+:::caution Rust support in Jupyter Notebooks is experimental. You might run into bugs, or unexpected behavior. :::
```
Comment (fnando) [inline docs/tools/developer-tools/jupyter-notebooks.mdx:14]: For some reason, all the content after this is wrapped by the message box. Maybe we need something a hard linebreak?

```suggestion
:::caution 
Rust support in Jupyter Notebooks is experimental. You might run into bugs, or unexpected behavior.
:::
```

**Review by janewang — APPROVED** [review]

File: docs/tools/developer-tools/jupyter-notebooks.mdx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+sidebar_position: 71
+---
+
+# Jupyter Notebooks
+
+Jupyter Notebooks are a document format that can include code, text, and other rich output. And with the appropriate setup they support Soroban Rust contracts.
+
+To use Soroban contracts in a Jupyter Notebook, the following setup is required. The following setup uses Visual Studio Code, but any Jupyter client and server can be used.
+
+:::caution
```
Comment (briwylde08) [inline docs/tools/developer-tools/jupyter-notebooks.mdx:14]: I think the Caution box isn't set up correctly. The preview has the entire article in the Caution box!

**Comment by janewang** [comment]
We typically put image in the asset folder: https://github.com/stellar/stellar-docs/commit/0a1cac665ca425b1a19b6644c86948ef0adde189

File: docs/tools/developer-tools/jupyter-notebooks.mdx (hunk 3 above)
Comment (ElliotFriend) [reply docs/tools/developer-tools/jupyter-notebooks.mdx:14]: Yes, it needs an empty line after the opening admonition tag, and before the closing tag. like this

```markdown
:::caution

Rust support in Jupyter ...

:::
```

**Pull request merged**